### DISCUSSION TITLE
Can external module use different version of Reanimated?

### DISCUSSION THREAD

**Discussion by piotrponikowski** [software-mansion/react-native-reanimated#5431] [Q&A] [answered]
Hello!
I am trying to upgrade reanimated and related modules to new versions. In my project I have:
```
"react-native-gesture-handler": "2.14.0",
"react-native-reanimated": "3.6.0",
"react-native-redash": "18.1.1",
"example-module": "1.0.0",
```
with this babel config:
```
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  plugins: ['react-native-reanimated/plugin'],
};
```

External example-module also requires Reanimated but in old version:
```
"react-native-gesture-handler": "1.10.3",
"react-native-reanimated": "1.13.4",
"react-native-redash": "14.2.2",
```

When I commented out all references to example-module, then everything works, however once I add import to example-module and try to rebuild app I receive:  `_TypeError: _ReanimatedModule.default.createNode is not a function (it is undefined), js engine: hermes_`

Does it mean external modules cannot use different Reanimated version, or something is wrong with my config and I should keep looking for solution?

**Comment by MatiPl01** [comment, marked as answer]
Yes, you should your project should use only one version of reanimated (so all libraries that depend on reanimated should be upgraded to versions which support your version of reanimated).

You can force the usage of the specific version of reanimated but it may break your dependencies if they rely on older/newer version of reanimated. More details in [docs](https://docs.swmansion.com/react-native-reanimated/docs/guides/troubleshooting/#multiple-versions-of-reanimated-were-detected).
